src/linear_combination.jl

```diff
@@ -61,10 +61,15 @@ function apply(table, LC::LinearCombination; cols=nothing)
 
     # Keep the generic form when not specifying column names
     # because that is much more performant than selecting each col by name
-    cols === nothing && return [_sum_row(row, LC.coefficients) for row in Tables.rows(table)]
+    result = if cols === nothing
+        _to_mat([_sum_row(row, LC.coefficients) for row in Tables.rows(table)])
```

Comment:
might just use `hcat` directly (c.f. `apply.jl`)

Reply:
In the end I just got rid of `_to_mat`. The code is getting quite messy and confusing so I'm cleaning it up in a follow up PR #63